tomviz/python/tomviz/web.py

```diff
@@ -0,0 +1,167 @@
+import os
+import shutil
+import zipfile
+
+from paraview import simple
+from paraview.web.dataset_builder import ImageDataSetBuilder
+from paraview.web.dataset_builder import CompositeDataSetBuilder
+
+
+def web_export(destinationPath, exportType, deltaPhi, deltaTheta):
+    dest = '%s/data' % destinationPath
+    camera = {
+        'type': 'spherical',
+        'phi': range(0, 360, deltaPhi),
+        'theta': range(-deltaTheta, deltaTheta + 1, deltaTheta)
+    }
+
+    # Copy application
+    copy_viewer(destinationPath)
+
+    # Choose export mode:
+    if exportType == 0:
+        export_images(dest, camera)
+
+    if exportType == 1:
+        export_layers(dest, camera)
+
+    # Zip data directory
+    zipData(destinationPath)
+
+# -----------------------------------------------------------------------------
+# Helpers
+# -----------------------------------------------------------------------------
+
+
+def zipData(destinationPath):
+    dstFile = os.path.join(destinationPath, 'data.tomviz')
+    dataDir = os.path.join(destinationPath, 'data')
+    zf = zipfile.ZipFile(dstFile, mode='w')
+    try:
+        for dirName, subdirList, fileList in os.walk(dataDir):
+            for fname in fileList:
+                fullPath = os.path.join(dirName, fname)
+                relPath = 'data/%s' % (os.path.relpath(fullPath, dataDir))
+                zf.write(fullPath, arcname=relPath,
+                         compress_type=zipfile.ZIP_STORED)
+    finally:
+        zf.close()
+
+    shutil.rmtree(dataDir)
+
+
+def get_proxy(id):
+    session = simple.servermanager.ActiveConnection.Session
+    remoteObj = session.GetRemoteObject(int(id))
+    return simple.servermanager._getPyProxy(remoteObj)
+
+
+def copy_viewer(destinationPath):
+    searchPath = os.getcwd()
+    for root, dirs, files in os.walk(searchPath):
+        if 'Tomviz.html' in files:
+            shutil.copy(os.path.join(root, 'Tomviz.html'), destinationPath)
+            return
+
+
+def add_scene_item(scene, name, proxy, view):
+    hasNormal = False
+    hasColor = False
+    colors = {}
+    representation = {}
+    rep = simple.GetRepresentation(proxy, view)
+
+    # Skip volume
+    if rep.Visibility == 0 or rep.Representation == 'Volume':
+        return
+
+    for prop in ['Representation']:
+        representation[prop] = rep.GetProperty(prop).GetData()
+
+    pdInfo = proxy.GetPointDataInformation()
+    numberOfPointArrays = pdInfo.GetNumberOfArrays()
+    for idx in xrange(numberOfPointArrays):
+        array = pdInfo.GetArray(idx)
+        rangeValues = array.GetRange(-1)
+        if array.Name == 'Normals':
+            hasNormal = True
```

Comment:
I don't agree as we are in a loop and we want to know if one of the array name was `Normals`.